### PR TITLE
Fix gcps validity check for older versions of rasterio

### LINKED ISSUE #464: BUG: empty gcps with older versions of rasterio (1.2.1)
See: https://github.com/corteva/rioxarray/actions/runs/1819395031

```
E           TypeError: Invalid value for attr 'gcps': {'type': 'FeatureCollection', 'features': []}. For serialization to netCDF files, its value must be of one of the following types: str, Number, ndarray, number, list, tuple
```

### PR DESCRIPTION
**Pull request by mraspaud**
- [ ] Closes #464 
 - [ ] Tests added
 - [ ] Fully documented, including `docs/history.rst` for all changes and `docs/rioxarray.rst` for new API

### CODE REVIEW

**Comment by mraspaud** [comment]
@snowman2 that seems to fix it.

File: rioxarray/_io.py [hunk 1]
```diff
@@ -880,7 +880,7 @@ def open_rasterio(
         coord_name = "band"
         coords[coord_name] = np.asarray(riods.indexes)
 
-    has_gcps = hasattr(riods, "gcps") and riods.gcps != ([], None)
+    has_gcps = hasattr(riods, "gcps") and riods.gcps[0] != []
```
Comment (snowman2) [inline rioxarray/_io.py:883]: Thoughts about this?
```suggestion
    has_gcps = hasattr(riods, "gcps") and riods.gcps[0]
```

Comment (mraspaud) [reply rioxarray/_io.py:883]: Fine by me!

**Review by snowman2 — APPROVED** [review]

**Comment by snowman2** [comment]
Thanks @mraspaud :+1:

**Comment by mraspaud** [comment]
My pleasure

**Pull request merged**